variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/AbstractVariant.java

```diff
@@ -181,11 +176,23 @@ public String getMainId() {
         return mainId;
     }
 
-    public void renormalizeVariant(long newStart, long newEnd, String newReference, String newAlternate) {
-        this.start = newStart;
-        this.end = newEnd;
-        this.reference = newReference;
-        this.alternate = newAlternate;
+    public void setReference(String reference) {
+        this.reference = (reference != null) ? reference : "";
+    }
+
+    public void setAlternate(String alternate) {
+        this.alternate = (alternate != null) ? alternate : "";
+    }
+
+    private void setCoordinates(long start, long end) {
+        if (start < 0 || end < 0) {
+            throw new IllegalArgumentException("Variant co-ordinate cannot be negative: " + this.toString());
```

Comment:
'coordinate' is a single word. Also calling `toString` from a constructor could lead to unexpected information being displayed, because the object might be partially constructed at that point. It would be preferable to show the negative coordinate instead.